backend/src/users/users.repository.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable } from '@nestjs/common';
+import { CreateUserDto } from './dto/create-user.dto';
+import { UpdateUserDto } from './dto/update-user.dto';
+import { PrismaService } from 'src/prisma/prisma.service';
+
+@Injectable()
+export class UsersRepository {
+  constructor(private readonly prisma: PrismaService) {}
+
+  async createSession(userId: number, token: string) {
```

Comment:
Quesito de estilo de código, mas ajuda bastante pra evitar de ter vários args em uma só função. Considere pra todos
```suggestion
  async createSession({userId, token}: {userId: number, token: string}) {
```